dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/ElasticManager.java

```diff
@@ -113,10 +112,10 @@ public void handleIndexUpdate(Entry entry, ElasticMode command) {
             switch (command) {
             case UPDATE:
                 post = restClient
-                        .performRequest("POST", "/entry/" + entryType + "/" + entry.getId() + "/_update", Collections.emptyMap(), entity);
+                        .performRequest("POST", "/entry/" + entryType + "/" + entry.getId() + "/_update" + "?refresh=wait_for", Collections.emptyMap(), entity);
```

Comment:
Ah, I see. 
Good find, but this might have performance implications if every PUT request needs to wait for a refresh to finish. 

Options 
1) test this PR with https://github.com/dockstore/dockstore-load-tester @coverbeck 
2) use the admin "refresh whole index" endpoint when the index needs to be refreshed synchronously for tests
http://odl-dyuen2.res.oicr.on.ca:8080/static/swagger-ui/index.html#/extendedGA4GH/toolsIndexGet

Other ideas welcome

Reply:
I'm happy to approach it in whichever way you find most expedient. In practice, having a delay on indexing new documents USUALLY doesn't present a problem to a client. Upon creation of some new item they are unlikely to go searching for it right away.

The refresh interval of ES defaults to 1 second, so unless there is a huge upsert your request responses aren't likely to ever exceed that. This takes some fiddling though, and `Thread.sleep` is a fine approach from a testing perspective if we want the ES Manager left untouched. 

The small number of documents also gives me some confidence that `wait_for` will not cause noticeable changes in user experience.

Reply:
My concern would be that someday (hopefully!) the number of documents won't be small, and then we'd have to remember to remove/change this.

Reply:
@coverbeck You're right 2) doesn't work because we're specifically testing an incremental update. I wonder if there's a direct API call that can be made to elastic search like "wait for refreshes that are in progress" for the tests

@david4096  If I understand correctly, I think that means PUT requests would be delayed for up to one second to wait for ES to trigger a refresh + the time to do the refresh. You're probably right that for now, that's probably not a big difference. 
But still, I think I'd rather have the test be a bit more ugly than delay all requests.

Reply:
@denis-yuen correct. You can change the refresh interval to tune to your use case https://www.elastic.co/guide/en/elasticsearch/reference/5.6/index-modules.html#dynamic-index-settings .

Further, you can manually trigger a refresh on an index using `POST /my_index/_refresh`. That is probably the best compromise between sleeping and waiting for refresh on every post. If I can figure out a straightforward way without adding much to ES manager we can go with that. Otherwise I'll try the thread sleeps.